lib/tasks/migrate_run_status.rake

```diff
@@ -0,0 +1,59 @@
+def record_previous_state(pr)
+  json_output = pr.to_json(include: [:pipeline_run_stages, :output_states])
+  file = Tempfile.new
+  file.write(json_output)
+  file.close
+  # Copy file to S3
+  pr_s3_file_name = "pipeline_run_#{pr.id}.json"
+  Open3.capture3("aws", "s3", "cp", file.path.to_s,
+                 "#{pr.archive_s3_path}/pre_migrate_run_status/#{pr_s3_file_name}")
+  file.unlink
+end
+
+def migrate_pre_run_stages(pr)
+  target_outputs = %w[ercc_counts taxon_counts taxon_byteranges]
+  if %w[CHECKED SUCCEEDED].include?(pr.job_status)
+    pr.update(results_finalized: PipelineRun::FINALIZED_SUCCESS)
+    target_outputs.each do |output|
+      OutputState.create(pipeline_run_id: pr.id, output: output, state: PipelineRun::STATUS_LOADED)
+    end
+  elsif %w[FAILED ERROR].include?(pr.job_status)
+    pr.create_output_states
+    pr.update(results_finalized: PipelineRun::FINALIZED_FAIL)
+  else
+    pr.create_output_states
+    pr.update(results_finalized: PipelineRun::IN_PROGRESS) # shouldn't be the case for any of those old runs
+  end
+  pr.create_run_stages
```

Comment:
I will probably move this to first line of the function.

Reply:
ok